protocol/contracts/beanstalk/field/FieldFacet.sol

```diff
@@ -167,89 +183,129 @@ contract FieldFacet is Invariable, ReentrancyGuard {
      */
     function _harvestPlot(
         address account,
+        uint256 fieldId,
         uint256 index
     ) private returns (uint256 harvestablePods) {
         // Check that `account` holds this Plot.
-        uint256 pods = s.accounts[account].field.plots[index];
+        uint256 pods = s.accounts[account].fields[fieldId].plots[index];
         require(pods > 0, "Field: no plot");
 
         // Calculate how many Pods are harvestable.
         // The upstream _harvest function checks that at least some Pods
         // are harvestable.
-        harvestablePods = s.field.harvestable.sub(index);
-        delete s.accounts[account].field.plots[index];
-
-        // Cancel any active Pod Listings active for this Plot.
-        // Note: duplicate of {Listing._cancelPodListing} without the
-        // ownership check, which is done above.
-        if (s.podListings[index] > 0) {
-            delete s.podListings[index];
-            emit PodListingCancelled(LibTractor._user(), index);
-        }
+        harvestablePods = s.fields[fieldId].harvestable.sub(index);
+
+        LibMarket._cancelPodListing(LibTractor._user(), fieldId, index);
+
+        delete s.accounts[account].fields[fieldId].plots[index];
 
         // If the entire Plot was harvested, exit.
         if (harvestablePods >= pods) {
             return pods;
         }
 
         // Create a new Plot with remaining Pods.
-        s.accounts[account].field.plots[index.add(harvestablePods)] = pods.sub(harvestablePods);
+        s.accounts[account].fields[fieldId].plots[index.add(harvestablePods)] = pods.sub(
+            harvestablePods
+        );
+    }
+
+    //////////////////// CONFIG /////////////////////
+
+    /**
+     * @notice Add a new Field to the system.
+     * @dev It is not possible to remove a Field, but a Field's Plan can be nullified.
+     */
+    function addField() public fundsSafu noSupplyChange noNetFlow {
+        LibDiamond.enforceIsOwnerOrContract();
+        uint256 fieldId = s.fieldCount;
+        s.fieldCount++;
+        emit FieldAdded(fieldId);
+    }
+
+    /**
+     * @notice Set the active Field. Only the active field is accrues Soil.
+     * @param fieldId ID of the Field to set as active. ID is the Field Number.
+     */
+    function setActiveField(uint256 fieldId, uint32 temperature) public fundsSafu noSupplyChange noNetFlow {
+        LibDiamond.enforceIsOwnerOrContract();
+        require(fieldId < s.fieldCount, "Field: Field does not exist");
+        s.activeField = fieldId;
+
+        // Reset weather.
```

Comment:
@Brean0 the weather and case system is a too much for me to dive into rn. does this make logic make sense for resetting the weather when updating fields?

Reply:
@pizzaman1337 may also have insight?